### ISSUE TITLE
RemoteSyslogLogger

### ISSUE THREAD

**Issue by mattclar** [papertrail/remote_syslog#90]
SO basically I am getting the following error but ONLY on my ActionCable server, not on my actual rails web server, Im not quite sure whats going on but it seems somewhat similar to Issue #6  ?? 
    ArgumentError: Tag may not contain spaces
    tag= at /usr/local/bundle/gems/syslog_protocol-0.9.2/lib/syslog_protocol/packet.rb:52
    initialize at /usr/local/bundle/gems/remote_syslog_logger-1.0.3/lib/remote_syslog_logger/udp_sender.rb:20
    new at /usr/local/bundle/gems/remote_syslog_logger-1.0.3/lib/remote_syslog_logger.rb:9

**Comment by troy** [comment]
How about some details? Like, say, what tag are you using, and does it contain spaces? :-P

**Comment by mattclar** [comment]
I honestly don't know what the tag is referring to? 
all I have set up for this gem is to add it to the gemfile and then add the following file to my config/environment.rb file:
      config.logger = RemoteSyslogLogger.new('logs3.papertrailapp.com', 21269)

as requested by the documentation
the strange thing is that the error doesn't occur with the rails server but only the ActionCable server even though they are running off of the same codebase

**Comment by troy** [comment]
Gotcha. It's called the `program` attribute here: https://github.com/papertrail/remote_syslog_logger#usage. If you instantiate `RemoteSyslogLogger` using the key and value in that example, it should start working. Here's an slightly representative example:

```
config.logger = RemoteSyslogLogger.new('logs3.papertrailapp.com', 12345, :program => "rails-#{RAILS_ENV}", :local_hostname => 'www42')
```

**Comment by angelacode** [comment]
I have the same problem, I don't use the :program attriute, what should go there, does it matter?

**Comment by angelacode** [comment]
here's the error I have:

/home/nitrous/code/site/.bundle/gems/syslog_protocol-0.9.2/lib/syslog_protocol/packet.rb:49:in `tag=': Tag must not be longer than 32 characters (ArgumentError)

**Comment by angelacode** [comment]
I have filenames > 32 characters -- I'd like to use the program name that is logging rather than overwriting it.

**Comment by troy** [comment]
@angelacode The 32 character limit is set by [RFC 3164](https://tools.ietf.org/html/rfc3164), the protocol which the [remote_syslog_logger](https://github.com/papertrail/remote_syslog_logger) library implements. Specifically:

> The TAG is a string of ABNF alphanumeric characters that MUST NOT exceed 32 characters.

**Comment by Rob117** [comment]
I ran into this limit 3 days into using the gem. I had a controller name that was a bit long (absolutely necessary), and this kicked off a deploy that caused a reboot cycle that ate a day of dev time.

I picked this because it was easy to install and setup, but I uninstalled it because I'm not willing to check that my filenames - starting from app/~ - are under 32 characters.

I respect that you are implementing the RFC, and that you are doing what you think is best, but this was a solid no for my usecase.
